gnome-initial-setup/pages/timezone/gis-timezone-page.c

```diff
@@ -78,13 +79,17 @@ typedef struct _GisTimezonePagePrivate GisTimezonePagePrivate;
 
 G_DEFINE_TYPE_WITH_PRIVATE (GisTimezonePage, gis_timezone_page, GIS_TYPE_PAGE);
 
+G_DEFINE_AUTO_CLEANUP_FREE_FUNC(locale_t, freelocale, NULL)
+
 static GDesktopClockFormat
 get_default_time_format (void)
 {
   const char *ampm, *nl_fmt;
+  const char *locale_str = setlocale (LC_ALL, NULL);
+  g_auto(locale_t) locale = newlocale (LC_ALL_MASK, locale_str, NULL);
 
   /* Default to 24 hour if we can't get the format from the locale */
-  nl_fmt = nl_langinfo (T_FMT);
+  nl_fmt = nl_langinfo_l (T_FMT, locale);
```

Comment:
@cosimoc I guess that's right, though I didn't see it failing without it. Ihave just updated the PR.